### PR TITLE
Add support for dots in globs (foo/bar/../bar/./boom.txt)

### PR DESCRIPTION
**Pull request by tokee**
It is fairly common to make relative paths to config files. This introduced a regression with the switch to globbing as default, as globbing did not support it. Hereby remedied.

### CODE REVIEW

File: src/main/java/dk/kb/util/Resolver.java [hunk 1]
```diff
@@ -116,10 +118,10 @@ public static URL resolveURL(String resourceName)
      * @return a list of Paths matching the given glob or the empty list if there were no matches.
      */
     public static List<Path> resolveGlob(String glob) {
-        List<Path> paths = new ArrayList<>();
         if (glob == null || glob.isEmpty()) {
-            return paths;
+            return Collections.emptyList();
```
Comment (sbktc) [inline src/main/java/dk/kb/util/Resolver.java:122]: Improvement to deliver an imutable empty list?

File: src/main/java/dk/kb/util/Resolver.java [hunk 2]
```diff
@@ -140,21 +142,55 @@ public static URL resolveURL(String resourceName)
...
+        int pos = 0;
+        while (pos < segments.size()-1) {
+            if (pos == 0 && "..".equals(segments.get(0))) { // Going further up than the root does not make sense
+                return Collections.emptyList();
+            }
+            if ("..".equals(segments.get(pos+1))) { // The subsequent segment goes one level up
+                segments.remove(pos+1);
+                segments.remove(pos);
+                pos = Math.max(0, pos-1); // Go one segment level back and try again
+                continue;
```
Comment (sbktc) [inline src/main/java/dk/kb/util/Resolver.java:187]: Given that the glob: "../foo/bar/baz/*.yaml" would yield a empty list, should the glob: "/foo/../../bar/baz/*.yaml" get the same treatment?

File: src/main/java/dk/kb/util/Resolver.java (hunk 1 above)
Comment (tokee) [reply src/main/java/dk/kb/util/Resolver.java:122]: Same behavior as before, I just found it to be cleaner code to delay the construction of a response structure until it has been checked whether there should be a "real" response or not. Also, explicitly using `Collections.emptyList();` is a very clear signal as to what is happening.

File: src/main/java/dk/kb/util/Resolver.java (hunk 2 above)
Comment (tokee) [reply src/main/java/dk/kb/util/Resolver.java:187]: The glob `../foo/bar/baz/.yaml` is not an "absolute glob". But if you mean `/../foo/bar/baz/.yaml` vs. `/foo/../..//foo/bar/baz/.yaml` then yes.

Comment (tokee) [reply src/main/java/dk/kb/util/Resolver.java:187]: Aaand I can see that I do not handle that properly in the code. I'll add a unit test and fix it!

Comment (sbktc) [reply src/main/java/dk/kb/util/Resolver.java:187]: Sorry, meant "/../foo/bar/baz/*.yaml".

Comment (tokee) [reply src/main/java/dk/kb/util/Resolver.java:191]: Done.

**Pull request merged**